lib/config.js

```diff
@@ -119,6 +120,11 @@ module.exports = {
 
       found = [].concat(found).map((conf) => prepare(conf));
 
+      if (mode) {
```

Comment:
not the right spot to do this. the flag value should be applied to the `result` in the `apply()` method of `lib/flags.js`. it's impossible to know which config in a multi-config that a user intends to apply a flag to, so it should apply to all configs.